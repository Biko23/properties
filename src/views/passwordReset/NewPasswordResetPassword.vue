<template>
  <v-container>
      <v-row>
          <v-col cols="12" md="6" offset-md="3">
              <v-row style="margin: 20vh auto;">
                  <v-col style="display:flex; align-items: center; justify-content: center;">
                       <v-form v-model="valid">
                            <v-container>
                                <p class="text-h4 text-center">Change Password</p>
                            <v-row>
                                <v-col cols="12" md="12">
                                    <v-row>
                                        <v-col cols="12" md="12">
                                            <v-text-field
                                                type="password"
                                                v-model="password"
                                                :rules="[rules.required, rules.min, rules.password]"
                                                label="New Password"
                                            ></v-text-field>
                                        </v-col>
                                        <v-col cols="12" md="12">
                                            <v-text-field
                                                type="password"
                                                v-model="confirmPassword"
                                                :rules="[(password === confirmPassword) || 'Password must match']"
                                                label="Confirm Password"
                                            ></v-text-field>
                                        </v-col>
                                        <v-col cols="12" md="12">
                                            <v-btn 
                                                block 
                                                color="#3B6EF3" 
                                                :disabled="!valid" 
                                                @click="changePassword"
                                            >
                                                <span style="color: white;">Reset Password</span>
                                            </v-btn>
                                        </v-col>
                                    </v-row>
                                </v-col>
                            </v-row>
                            </v-container>
                        </v-form>
                  </v-col>
              </v-row>
          </v-col>
      </v-row>
  </v-container>
</template>
<script>

import { mapActions } from 'vuex';
export default {
  name: 'NewPasswordResetPassword',
  data(){
      return {
          valid: true,
          password: '',
          confirmPassword: '',
          rules: {
            required: value => !!value || "Required.",
            min: v => (v && v.length >= 6) || "Min 8 characters",
            password: v => /(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])/.test(v) || "Password should have an upper case, lower case and letters"
        }
      }
  },
  methods: {
      ...mapActions(['resetPassword']),
      changePassword(){
          alert(JSON.stringify({password: this.password, confirm: this.confirmPassword}))
      },
      async changePassword(){
          try {
              const response = await this.resetPassword(this.password);
              if(response.data.status == 1){
                  alert(JSON.stringify(response.data.message))
                  setTimeout(() => {
                    this.$router.push('/login');
                  }, 2000);
              } else {
                  alert(JSON.stringify(response.data.message));
                  setTimeout(() => {
                    this.$router.push('/');
                  }, 2000);
              }
          } catch (error) {
              console.log(error.message);
          }
      }
  }
}
</script>

<style>

</style>