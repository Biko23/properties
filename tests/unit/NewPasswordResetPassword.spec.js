import { createLocalVue, mount } from '@vue/test-utils'
import NewPasswordResetPassword from '@/views/passwordReset/NewPasswordResetPassword.vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'


describe('NewPasswordResetPassword.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    let actions
    let store
    let vuetify

    beforeEach(() => {
        vuetify = new Vuetify()
        actions = {
            resetPassword: jest.fn()
        }
        store = new Vuex.Store({
            actions
        })
    })

    const mountFactory = options => {
        return mount(NewPasswordResetPassword, {
            localVue,
            vuetify,
            store,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFactory({})
        expect(wrapper.html()).toMatchSnapshot()
    })
})

