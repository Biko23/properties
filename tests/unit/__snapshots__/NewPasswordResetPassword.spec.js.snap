// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPasswordResetPassword.vue should have a match snapshot 1`] = `
<div class="container">
  <base-dialog message="" title=""></base-dialog>
  <div class="row">
    <div class="col-md-6 offset-md-3 col-12">
      <div class="row" style="margin: 20vh auto;">
        <div class="col" style="display: flex; align-items: center; justify-content: center;">
          <form novalidate="novalidate" class="v-form">
            <div class="container">
              <p class="text-h4 text-center">Change Password</p>
              <div class="row">
                <div class="col-md-12 col-12">
                  <div class="row">
                    <div class="col-md-12 col-12">
                      <div class="v-input theme--light v-text-field">
                        <div class="v-input__control">
                          <div class="v-input__slot">
                            <div class="v-text-field__slot"><label for="input-5" class="v-label theme--light" style="left: 0px; position: absolute;">New Password</label><input id="input-5" type="password"></div>
                          </div>
                          <div class="v-text-field__details">
                            <div class="v-messages theme--light">
                              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div class="col-md-12 col-12">
                      <div class="v-input theme--light v-text-field">
                        <div class="v-input__control">
                          <div class="v-input__slot">
                            <div class="v-text-field__slot"><label for="input-8" class="v-label theme--light" style="left: 0px; position: absolute;">Confirm Password</label><input id="input-8" type="password"></div>
                          </div>
                          <div class="v-text-field__details">
                            <div class="v-messages theme--light">
                              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div class="col-md-12 col-12" style="display: flex; justify-content: space-evenly;"><button type="button" class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light v-size--default" style="background-color: rgb(59, 110, 243); border-color: #3b6ef3;"><span class="v-btn__content"><span style="color: white;">Reset Password</span></span></button></div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
